osu.Game.Rulesets.Osu.Tests/TestCaseSliderInput.cs

```diff
@@ -0,0 +1,240 @@
+            AddStep("Tracking retention test", () =>
+            {
+                var actions = new List<List<OsuAction>>
+                {
+                    new List<OsuAction> { OsuAction.LeftButton, OsuAction.RightButton },
+                    new List<OsuAction> { OsuAction.RightButton }
+                };
+
+                performStaticInputTest(actions, true);
+            });
+
+            AddUntilStep(() => allJudgedFired, "Wait for test 4");
+            AddAssert("Tracking retained", assertGreatJudge);
+        }
+
+        private bool assertMehJudge()
+        {
+            return lastResult.Type == HitResult.Meh;
+        }
+
+        private bool assertGreatJudge()
+        {
+            return lastResult.Type == HitResult.Great;
+        }
+
+        private void performStaticInputTest(List<List<OsuAction>> actionsOnSlider, bool primeKey = false)
+        {
+            var slider = new Slider
+            {
+                StartTime = 1500,
+                Position = new Vector2(100, 100),
+                Path = new SliderPath(PathType.PerfectCurve, new[]
+                {
+                    Vector2.Zero,
+                    new Vector2(5, 0),
+                }, 5),
+            };
+
+            var frames = new List<ReplayFrame>();
+            double frameTime = 0;
+
+            // Empty frame to be added as a workaround for first frame behavior.
+            // If an input exists on the first frame, the input would apply to the entire intro lead-in
+            // Likely requires some discussion regarding how first frame inputs should be handled.
+            frames.Add(new OsuReplayFrame
+            {
+                Position = slider.Position,
+                Time = 0,
+                Actions = new List<OsuAction>()
+            });
+
+            frames.Add(new OsuReplayFrame
+            {
+                Position = slider.Position,
+                Time = 250,
+                Actions = primeKey ? new List<OsuAction> { OsuAction.LeftButton } : new List<OsuAction>()
+            });
+
+            foreach (var a in actionsOnSlider)
+            {
+                frames.Add(new OsuReplayFrame
+                {
+                    Position = slider.Position,
+                    Time = slider.StartTime + frameTime,
+                    Actions = a
+                });
+                frameTime += 250;
+            }
+
+            Beatmap.Value = new TestWorkingBeatmap(new Beatmap<OsuHitObject>
+            {
+                HitObjects = { slider },
+                ControlPointInfo = new ControlPointInfo { DifficultyPoints = { new DifficultyControlPoint { SpeedMultiplier = 0.1f } } },
+                BeatmapInfo = new BeatmapInfo { BaseDifficulty = new BeatmapDifficulty { SliderTickRate = 3 }, Ruleset = new OsuRuleset().RulesetInfo },
+            });
+
+            ScoreAccessibleReplayPlayer player = new ScoreAccessibleReplayPlayer(new Score { Replay = new Replay { Frames = frames } })
```

Comment:
```suggestion
            var player = new ScoreAccessibleReplayPlayer(new Score { Replay = new Replay { Frames = frames } })
```

this is a good example of where `var` should be used.